### PR TITLE
feat: compute schematic id only from the extensions

### PR DESCRIPTION
**Pull request by utkuozdemir**
When determining the schematic ID of a machine, instead of relying the ID on the schematic ID meta-extension, compute the ID by gathering the extensions on the machine. This way, the extension ID will not contain the META values, labels or the kernel args.

This ID is actually the ID we need, as when we compare the desired schematic with the actual one during a Talos upgrade, we are only interested in the changes in the list of extensions.

This does not cause the kernel args, labels, etc. to disappear, as they are used at installation time and preserved afterward (e.g., during upgrades).

Additionally:
- Remove the list of extensions from the `Schematic` resource, as it relied upon the schematics always being created through Omni. This is not always the case - i.e., when a partial join config is used. Therefore, instead of relying on it, we store the list of extensions by directly reading them from the machine and storing them on the `MachineStatus` resource.
- Skip setting the schematic META section at all if there are no labels set on Download Installation Media screen.

Closes siderolabs/omni#55.

### CODE REVIEW

File: internal/backend/runtime/omni/controllers/omni/talos_extensions.go [hunk 1]
```diff
@@ -30,7 +29,7 @@ import (
 type TalosExtensionsController = qtransform.QController[*omni.TalosVersion, *omni.TalosExtensions]
 
 // NewTalosExtensionsController instantiates the TalosExtensions controller.
-func NewTalosExtensionsController() *TalosExtensionsController {
+func NewTalosExtensionsController(imageFactoryClient *imagefactory.Client) *TalosExtensionsController {
```
Comment (utkuozdemir) [inline internal/backend/runtime/omni/controllers/omni/talos_extensions.go:32]: Switch to using the new wrapper, so the only point where we create a new image factory GRPC client is actually in `magefactory.Client`

File: frontend/src/views/omni/Modals/DownloadInstallationMedia.vue [hunk 2]
```diff
@@ -311,7 +311,7 @@ const createSchematic = async () => {
       meta_values: {},
     };
 
-    if (labels.value) {
+    if (labels.value && Object.keys(labels.value).length > 0) {
```
Comment (utkuozdemir) [inline frontend/src/views/omni/Modals/DownloadInstallationMedia.vue:314]: If no labels were set, we were setting here a YAML with empty content for the labels key. Now we do not set any META information if there are no labels.

File: internal/backend/runtime/omni/controllers/omni/cluster_machine_config.go [hunk 3]
```diff
@@ -284,7 +284,7 @@ func (clusterMachineConfigControllerHelper) generateConfig(
 		return nil, fmt.Errorf("talos version is not set on the resource %s", clusterConfigVersion.Metadata())
 	}
 
-	installImage, err := buildInstallImage(clusterMachineTalosVersion, machineStatus, talosVersion)
+	installImage, err := buildInstallImage(clusterMachineTalosVersion, machineStatus, talosVersion) // here should come from the full schematic id, with args
```
Comment (Unix4ever) [inline internal/backend/runtime/omni/controllers/omni/cluster_machine_config.go:287]: this comment should be dropped I guess

File: internal/backend/runtime/omni/controllers/omni/cluster_machine_config_status.go [hunk 4]
```diff
@@ -345,17 +345,18 @@ func (h *clusterMachineConfigStatusControllerHandler) syncTalosVersionAndSchemat
...
+
 	if actualVersion == expectedVersion && actualSchematic == expectedSchematic {
 		configStatus.TypedSpec().Value.TalosVersion = actualVersion
 		configStatus.TypedSpec().Value.SchematicId = actualSchematic
 
 		return true, nil
 	}
 
-	image, err := buildInstallImage(talosVersion, machineStatus, expectedVersion)
+	image, err := buildInstallImage(talosVersion, machineStatus, expectedVersion) // this should come "plain", without kernel args
```
Comment (Unix4ever) [inline internal/backend/runtime/omni/controllers/omni/cluster_machine_config_status.go:359]: this comment too

File: internal/backend/runtime/omni/controllers/omni/internal/talos/schematic.go [hunk 5]
```diff
@@ -20,30 +20,60 @@ import (
...
-// extension, or calculates vanilla schematic ID if there's none.
-func GetSchematicID(ctx context.Context, c *client.Client) (string, error) {
-	extensions := map[resource.ID]*runtime.ExtensionStatus{}
+// SchematicInfo contains the information about the schematic - the plain schematic ID and the extensions.
+type SchematicInfo struct {
+	ID         string
+	Extensions []string
+}
+
+// GetSchematicInfo uses Talos API list all the schematics, and computes the plain schematic ID,
```
Comment (Unix4ever) [inline internal/backend/runtime/omni/controllers/omni/internal/talos/schematic.go:29]:
```suggestion
// GetSchematicInfo uses Talos API to list all the schematics, and computes the plain schematic ID,
```

**Review by Unix4ever — APPROVED** [review]

File: frontend/src/api/resources.ts [hunk 6]
```diff
@@ -150,7 +150,7 @@ export const KubernetesUsageType = "KubernetesUsages.omni.sidero.dev";
 export const PermissionsID = "permissions";
 export const PermissionsType = "Permissions.omni.sidero.dev";
 export const SecureBoot = "secureboot";
-export const DefaultTalosVersion = "1.6.4";
+export const DefaultTalosVersion = "1.6.6";
```
Comment (frezbo) [inline frontend/src/api/resources.ts:153]: 1.6.7

**Review by frezbo — APPROVED** [review]

**Comment by utkuozdemir** [comment]
🙏🏻

**Pull request merged**